database/raw.sql

```diff
@@ -0,0 +1,1590 @@
+        ev.screen_view__transition_type as last_screen_view__transition_type,
+        ev.screen_view__type as last_screen_view__type,
+        session_identifier
+    from unified_events ev
+      LEFT JOIN
+    unified_dim_rfc_5646_language_mapping l ON lower(ev.br_lang) = lower(l.lang_tag)
+      LEFT JOIN
+    unified_dim_geo_country_mapping g ON lower(ev.geo_country) = lower(g.alpha_2)
+    where
+        event_name in ('page_view', 'screen_view')
+        and view_id is not null
+    and coalesce(iab__spider_or_robot, False ) = False
+  and not useragent not similar to '.*(bot|crawl|slurp|spider|archiv|spinn|sniff|seo|audit|survey|pingdom|worm|capture|(browser|screen)shots|analyz|index|thumb|check|facebook|PingdomBot|PhantomJS|YandexBot|Twitterbot|a_archiver|facebookexternalhit|Bingbot|BingPreview|Googlebot|Baiduspider|360(Spider|User-agent)|semalt).*'
+      qualify row_number() over (partition by session_identifier order by derived_tstamp desc) = 1
+    
+),
+session_aggs as (
+    select
+      session_identifier
+      , min(derived_tstamp) as start_tstamp
+      , max(derived_tstamp) as end_tstamp
+      , count(*) as total_events
+      , count(distinct view_id) as views
+          , 
+      histogram(event_name) AS event_counts
+      ,JSON(histogram(event_name)) as event_counts_string,
+     (GETVARIABLE('heartbeat') * (
+              -- number of (unqiue in heartbeat increment) pages pings following a page ping (gap of heartbeat)
+              count(distinct case when event_name = 'page_ping' and view_id is not null then
+                          -- need to get a unique list of floored time PER page view, so create a dummy surrogate key...
+                          view_id || cast(floor(epoch(dvce_created_tstamp)/GETVARIABLE('heartbeat')) as TEXT)
+                      else null end) - count(distinct case when event_name = 'page_ping' and view_id is not null then view_id else null end)
+                            ))  +
+                            -- number of page pings following a page view (or no event) (gap of min visit length)
+                            (count(distinct case when event_name = 'page_ping' and view_id is not null then view_id else null end) * GETVARIABLE('min_visit_length')) as engaged_time_in_s_web
+      , datediff('second', min(derived_tstamp), max(derived_tstamp)) as absolute_time_in_s
+    from unified_events
+    where 1 = 1
+    and coalesce(iab__spider_or_robot, False ) = False
+  and useragent not similar to '.*(bot|crawl|slurp|spider|archiv|spinn|sniff|seo|audit|survey|pingdom|worm|capture|(browser|screen)shots|analyz|index|thumb|check|facebook|PingdomBot|PhantomJS|YandexBot|Twitterbot|a_archiver|facebookexternalhit|Bingbot|BingPreview|Googlebot|Baiduspider|360(Spider|User-agent)|semalt).*'
+    group by session_identifier
+),
+session_aggs_with_engaged_time as (
+    select a.*
+      , a.engaged_time_in_s_web as engaged_time_in_s
+    from session_aggs a
+),
+iso_639_3_deduped AS (
```

Comment:
Should you just do this once when you import the CSV?

Reply:
Probably - but I've copied like for like what is in our current unified models.